### PR TITLE
Support k8s 1.22 and update dependencies

### PR DESCRIPTION
**Pull request by d-kuro**
refs: https://github.com/cybozu-go/moco/issues/332

### CODE REVIEW

File: .cr.yaml [hunk 1]
```diff
@@ -1,4 +0,0 @@
-# This file is the config file for helm/chart-releaser
-owner: cybozu-go
-git-repo: moco
-release-name-template: "{{ .Name }}-chart-v{{ .Version }}"
```
Comment (d-kuro) [inline .cr.yaml:4]: This file was no longer needed.

File: ct.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # This file is the config file for helm/chart-testing
 target-branch: main
 validate-maintainers: false
-check-version-increment: true
+check-version-increment: false
```
Comment (d-kuro) [inline ct.yaml:4]: Chart releases are now done manually
Version increments are done at arbitrary times.
I disabled CI checks.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -47,7 +47,7 @@ jobs:
     strategy:
       matrix:
         mysql-version: ["8.0.27"]
-        k8s-version: ["1.19.11", "1.20.7", "1.21.1"]
+        k8s-version: ["1.20.7", "1.21.1", "1.22.2"]
```
Comment (zoetrope) [inline .github/workflows/ci.yaml:50]:
```suggestion
        k8s-version: ["1.20.7", "1.21.2", "1.22.2"]
```
Makefile uses `1.21.2`.

**Review by zoetrope — CHANGES_REQUESTED** [review]
I checked the changes in kubebuilder 3.2.0.
Please fix the followings.

- admissionReviewVersions should only support v1.
  - https://github.com/cybozu-go/moco/blob/04d00bbee82abccc907e69bad763d8b303bc330e/api/v1beta1/mysqlcluster_webhook.go#L25
  - https://github.com/cybozu-go/moco/blob/04d00bbee82abccc907e69bad763d8b303bc330e/api/v1beta1/mysqlcluster_webhook.go#L44
  - https://github.com/cybozu-go/moco/blob/3288f037e4efc1c8a7a041b88b90dce5d1f61f8d/api/v1beta1/backuppolicy_webhook.go#L26
- Please add `kubectl.kubernetes.io/default-container: manager` annotation to manager.yaml.
  - https://github.com/kubernetes-sigs/kubebuilder/blob/b6c36fa9eecb5f72680e7b8acf26cfc7a7c2fb50/pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/manager/config.go#L69
- Call `defer GinkgoRecover()` webhook_suite_test.go.
  - https://github.com/kubernetes-sigs/kubebuilder/pull/2345

**Comment by zoetrope** [comment]
Should we still support admissionReviewVersions/v1beta1?

**Review by masa213f — COMMENTED** [review]
I don't have any other comment. Please reflect zoetrope's comments.

**Comment by d-kuro** [comment]
@zoetrope Thanks for reviewing it.
I Fixed:

* Call `GinkgoRecover`: https://github.com/cybozu-go/moco/pull/338/commits/57ee77a8170e9cb7ff893067f79fcbc0525b99cf
* Remove v1beta1 from admissionReviewVersions: https://github.com/cybozu-go/moco/pull/338/commits/c02b6d6d1b7dd649c01301468e8c76bedeedc9a6
* Add `default-container` annotation: https://github.com/cybozu-go/moco/pull/338/commits/16ca204da833f8bba0c0cfdc5fe2fd140f2a2902
* Use Kubernetes v1.21.2 for e2e tests: https://github.com/cybozu-go/moco/pull/338/commits/542aa901bbe6200ca14763a439a83d407fe45ed5

> Should we still support admissionReviewVersions/v1beta1?

I didn't think it necessary.

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**